### PR TITLE
dgedit: init at 0.10.0

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by puzzlewolf — CHANGES_REQUESTED** [review]
Doesn't work for me:
```
[nix-shell:~/.cache/nixpkgs-review/pr-83597]$ ./results/dgedit/bin/dgedit 
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Aborted (core dumped)
```
Perhaps a missing qt dependency?

**Comment by matthiasbeyer** [comment]
Oh, yeah, I did not finish this...

**Comment by matthiasbeyer** [comment]
So, as far as I can see, this is built withg qt4. I still get the following error:

```
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```

And a segfault afterwards. But I am not sure whether this is due that I run wayland and sway and whether it would work on x11...

File: pkgs/applications/audio/dgedit/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ stdenv
+, fetchgit
+, pkgconfig
```
Comment (jonringer) [inline pkgs/applications/audio/dgedit/default.nix]: canonical name over alias
```suggestion
, pkg-config
```

File: pkgs/applications/audio/dgedit/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  version = "0.10.0";
+
+  src = fetchgit {
+    url = "git://git.drumgizmo.org/dgedit.git/";
+    rev = "v${version}";
+    sha256 = "11mcfa6h61q0gk0ynfpqql4lnih0m3y2z9mdjq4p1219hrapxs5q";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkgconfig autoreconfHook ];
```
Comment (jonringer) [inline pkgs/applications/audio/dgedit/default.nix]:
```suggestion
  nativeBuildInputs = [ pkg-config autoreconfHook ];
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

File: pkgs/applications/audio/dgedit/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    rev = "v${version}";
+    sha256 = "11mcfa6h61q0gk0ynfpqql4lnih0m3y2z9mdjq4p1219hrapxs5q";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkg-config autoreconfHook ];
+  buildInputs = [ qt4 libao libsndfile ];
+
+  meta = with stdenv.lib; {
+    description = "the DrumGizmo drumkit editor";
```
Comment (jonringer) [inline pkgs/applications/audio/dgedit/default.nix]: nitpick, should have caught this last time
```suggestion
    description = "The DrumGizmo drumkit editor";
```

**Comment by matthiasbeyer** [comment]
Ready to rebase?

**Review by jonringer — CHANGES_REQUESTED** [review]
doesn't build for me
```
builder for '/nix/store/6vvwnrfrlzk2qlvv00rcqy4i6jx8mymq-dgedit.drv' failed with exit code 1; last 10 log lines:
  fatal: unable to connect to git.drumgizmo.org:
  git.drumgizmo.org[0: 212.178.176.2]: errno=Connection refused

  fatal: unable to connect to git.drumgizmo.org:
  git.drumgizmo.org[0: 212.178.176.2]: errno=Connection refused

  fatal: unable to connect to git.drumgizmo.org:
  git.drumgizmo.org[0: 212.178.176.2]: errno=Connection refused

  Unable to checkout refs/tags/v0.10.0 from git://git.drumgizmo.org/dgedit.git/.
```

**Review by jonringer — CHANGES_REQUESTED** [review]
please fix:
```
$ ./results/dgedit/bin/dgedit
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Aborted (core dumped)
```
https://nixos.org/nixpkgs/manual/#sec-language-qt

**Comment by puzzlewolf** [comment]
The binary works on X for me now :)

File: pkgs/applications/audio/dgedit/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkgconfig autoreconfHook wrapQtAppsHook ];
+  buildInputs = [ qtbase qtx11extras qttools libao libsndfile ];
+
+  meta = with stdenv.lib; {
+    description = "the DrumGizmo drumkit editor";
+    homepage = "https://drumgizmo.org/";
+    license = licenses.gpl3Plus;
```
Comment (puzzlewolf) [inline pkgs/applications/audio/dgedit/default.nix:30]: The licensing situation of this project seems a bit mixed. The program (menu Help/about/License) displays a  GPL 3, calls it GPL (without version number), but the source files say "GPL version 2 or later" :confused:. Obviously, GPL 3 is included in "or later". I don't know what we should do here, maybe ask upstream to clarify? Is this actually a problem?

File: pkgs/applications/audio/dgedit/default.nix (hunk 3 above)
Comment (puzzlewolf) [reply pkgs/applications/audio/dgedit/default.nix:28]: Did you keep the small `the` on purpose?

**Review by puzzlewolf — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/dgedit/default.nix (hunk 4 above)
Comment (matthiasbeyer) [reply pkgs/applications/audio/dgedit/default.nix:30]: Clearification would be nice, yes.

File: pkgs/applications/audio/dgedit/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  version = "0.10.0";
+
+  src = fetchgit {
+    url = "git://git.drumgizmo.org/dgedit.git/";
+    rev = "v${version}";
+    sha256 = "11mcfa6h61q0gk0ynfpqql4lnih0m3y2z9mdjq4p1219hrapxs5q";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkgconfig autoreconfHook wrapQtAppsHook ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/dgedit/default.nix:24]:
```suggestion
  nativeBuildInputs = [ pkg-config autoreconfHook wrapQtAppsHook ];
```

File: pkgs/applications/audio/dgedit/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{ stdenv
+, fetchgit
+, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/dgedit/default.nix:3]:
```suggestion
, pkg-config
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 83597` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dgedit</li>
  </ul>
</details>

**Comment by matthiasbeyer** [comment]
Nice, thanks.
Btw: LOL github... does not even make the default commit message a real fixup message. :angry:

**Review by SuperSandro2000 — COMMENTED** [review]
Can you squash the commits?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 83597` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dgedit</li>
  </ul>
</details>

**Comment by matthiasbeyer** [comment]
Done

File: pkgs/applications/audio/dgedit/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    url = "git://git.drumgizmo.org/dgedit.git/";
+    rev = "v${version}";
+    sha256 = "11mcfa6h61q0gk0ynfpqql4lnih0m3y2z9mdjq4p1219hrapxs5q";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkg-config autoreconfHook wrapQtAppsHook ];
+  buildInputs = [ qtbase qtx11extras qttools libao libsndfile ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/dgedit/default.nix:27]:
```suggestion
  meta = with lib; {
```

File: pkgs/applications/audio/dgedit/default.nix [hunk 8]
```diff
@@ -0,0 +1,34 @@
+{ stdenv
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/dgedit/default.nix:1]:
```suggestion
{ lib
```

File: pkgs/applications/audio/dgedit/default.nix [hunk 9]
```diff
@@ -0,0 +1,34 @@
+{ lib
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/dgedit/default.nix:1]:
```suggestion
{ stdenv
, lib
```

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM. Can someone with X11 confirm that it does not segfault?

File: pkgs/applications/audio/dgedit/default.nix [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "v${version}";
+    sha256 = "11mcfa6h61q0gk0ynfpqql4lnih0m3y2z9mdjq4p1219hrapxs5q";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkg-config autoreconfHook wrapQtAppsHook ];
+  buildInputs = [ qtbase qtx11extras qttools libao libsndfile ];
+
+  meta = with lib; {
+    description = "the DrumGizmo drumkit editor";
```
Comment (siraben) [inline pkgs/applications/audio/dgedit/default.nix:29]: Please capitalize "the" or remove it.

**Review by siraben — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/dgedit/default.nix (hunk 10 above)
Comment (matthiasbeyer) [reply pkgs/applications/audio/dgedit/default.nix:29]: Feel free to send a patch.

Comment (SuperSandro2000) [reply pkgs/applications/audio/dgedit/default.nix:29]:
```suggestion
    description = "The DrumGizmo drumkit editor";
```

**Comment by siraben** [comment]
Build failure on darwin
```
make[1]: Entering directory '/private/tmp/nix-build-dgedit-0.10.0.drv-0/dgedit'
make[2]: Entering directory '/private/tmp/nix-build-dgedit-0.10.0.drv-0/dgedit'
make[2]: Nothing to be done for 'install-exec-am'.
make[2]: Nothing to be done for 'install-data-am'.
make[2]: Leaving directory '/private/tmp/nix-build-dgedit-0.10.0.drv-0/dgedit'
make[1]: Leaving directory '/private/tmp/nix-build-dgedit-0.10.0.drv-0/dgedit'
qtOwnPathsHook
post-installation fixup
wrapping Qt applications in /nix/store/wazwcnh64pcccnrblz92bn2gk8wqa29i-dgedit-0.10.0/bin /nix/store/wazwcnh64pcccnrblz92bn2gk8wqa29i-dgedit-0.10.0/sbin /nix/store/wazwcnh64pcccnrblz92bn2gk8wqa29i-dgedit-0.10.0/libexec
strip is /nix/store/ynbk6s6nqmrvv17snwydkfn64h81r4yv-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/wazwcnh64pcccnrblz92bn2gk8wqa29i-dgedit-0.10.0/bin 
patching script interpreter paths in /nix/store/wazwcnh64pcccnrblz92bn2gk8wqa29i-dgedit-0.10.0
postPatchMkspecs
postPatchMkspecs
postPatchMkspecs
/nix/store/wazwcnh64pcccnrblz92bn2gk8wqa29i-dgedit-0.10.0
qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

[1]    47483 abort      ./result/bin/dgedit
```
Looks like it could be a trivial fix.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 83597` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>dgedit</li>
</ul>
</details>

**Pull request closed without merging**